INFO 2023-09-12 08:12:12,158 Watching for file changes with StatReloader
INFO 2023-09-12 08:12:18,909 "GET / HTTP/1.1" 302 0
INFO 2023-09-12 08:12:19,426 "GET /accounts/login/?next=/ HTTP/1.1" 200 3985
INFO 2023-09-12 08:12:19,492 "GET /static/plugins/fontawesome-free/css/all.min.css HTTP/1.1" 200 58578
INFO 2023-09-12 08:12:19,492 "GET /static/plugins/toastr/toastr.min.css HTTP/1.1" 200 6454
INFO 2023-09-12 08:12:19,493 "GET /static/plugins/overlayScrollbars/css/OverlayScrollbars.min.css HTTP/1.1" 200 19724
INFO 2023-09-12 08:12:19,495 "GET /static/dist/css/login-page.css HTTP/1.1" 200 8610
INFO 2023-09-12 08:12:19,496 "GET /static/dist/css/adminlte.min.css HTTP/1.1" 200 686707
INFO 2023-09-12 08:12:19,530 "GET /static/plugins/toastr/toastr.min.js HTTP/1.1" 200 5251
INFO 2023-09-12 08:12:19,531 "GET /static/dist/js/jquery.formset.js HTTP/1.1" 200 13632
INFO 2023-09-12 08:12:19,531 "GET /static/plugins/bootstrap/js/bootstrap.bundle.min.js HTTP/1.1" 200 80698
INFO 2023-09-12 08:12:19,531 "GET /static/plugins/overlayScrollbars/js/jquery.overlayScrollbars.min.js HTTP/1.1" 200 42383
INFO 2023-09-12 08:12:19,532 "GET /static/plugins/jquery/jquery.min.js HTTP/1.1" 200 88145
INFO 2023-09-12 08:12:19,535 "GET /static/dist/js/demo.js HTTP/1.1" 200 12616
INFO 2023-09-12 08:12:19,536 "GET /static/dist/js/adminlte.js HTTP/1.1" 200 57530
INFO 2023-09-12 08:12:19,537 "GET /static/dist/img/login.png HTTP/1.1" 200 144918
INFO 2023-09-12 08:12:19,863 "GET /static/plugins/fontawesome-free/webfonts/fa-solid-900.woff2 HTTP/1.1" 200 79444
WARNING 2023-09-12 08:12:21,781 Not Found: /favicon.ico
WARNING 2023-09-12 08:12:21,781 "GET /favicon.ico HTTP/1.1" 404 8074
INFO 2023-09-12 08:12:37,045 "POST /users/accounts/login/ HTTP/1.1" 302 0
INFO 2023-09-12 08:12:38,015 "GET / HTTP/1.1" 200 9842
INFO 2023-09-12 08:12:38,076 "GET /static/plugins/Datatables/datatables.min.css HTTP/1.1" 200 25780
INFO 2023-09-12 08:12:38,080 "GET /static/plugins/Datatables/datatables.min.js HTTP/1.1" 200 2042446
INFO 2023-09-12 08:12:38,822 "GET /static/plugins/fontawesome-free/webfonts/fa-regular-400.woff2 HTTP/1.1" 200 13584
INFO 2023-09-12 08:12:46,556 "GET /session/list/ HTTP/1.1" 200 12150
INFO 2023-09-12 08:12:53,143 "GET /session/list/ HTTP/1.1" 200 12150
INFO 2023-09-12 08:12:55,630 "GET /session/list/ HTTP/1.1" 200 12150
INFO 2023-09-12 08:13:08,372 "GET /semester/list/ HTTP/1.1" 200 12859
INFO 2023-09-12 08:13:29,268 "GET /cohort/list/ HTTP/1.1" 200 12553
INFO 2023-09-12 08:13:37,328 "GET /users/list/ HTTP/1.1" 200 11183
INFO 2023-09-12 08:13:45,146 "GET /student/list HTTP/1.1" 200 12017
INFO 2023-09-12 08:13:48,374 "GET /staff/list/ HTTP/1.1" 200 11320
INFO 2023-09-12 08:14:46,737 "GET /staff/list/ HTTP/1.1" 200 11320
ERROR 2023-09-12 08:15:09,515 Internal Server Error: /result/view/all
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/list.py", line 174, in get
    context = self.get_context_data()
  File "/home/lanlokun/Desktop/portal/apps/result/views.py", line 26, in get_context_data
    context["courses"] = Course.objects.filter(
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/query.py", line 1436, in filter
    return self._filter_or_exclude(False, args, kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/query.py", line 1454, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/query.py", line 1461, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1534, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1565, in _add_q
    child_clause, needed_inner = self.build_filter(
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1415, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1225, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1713, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'semester' into field. Choices are: code, coursematerial, credit_unit, id, lecturer, lecturer_id, level, name, result, studentcourse
ERROR 2023-09-12 08:15:09,517 "GET /result/view/all HTTP/1.1" 500 135562
INFO 2023-09-12 08:15:20,580 "GET /users/accounts/logout/ HTTP/1.1" 302 0
INFO 2023-09-12 08:15:20,781 "GET / HTTP/1.1" 302 0
INFO 2023-09-12 08:15:21,303 "GET /accounts/login/?next=/ HTTP/1.1" 200 3985
INFO 2023-09-12 08:15:43,331 "POST /users/accounts/login/ HTTP/1.1" 302 0
INFO 2023-09-12 08:15:44,299 "GET / HTTP/1.1" 200 6540
INFO 2023-09-12 08:15:47,511 "GET /staff/class/ HTTP/1.1" 200 8010
INFO 2023-09-12 08:15:50,457 "GET /staff/class/6/assignment HTTP/1.1" 200 10697
INFO 2023-09-12 08:15:59,436 "GET /staff/class/ HTTP/1.1" 200 8010
INFO 2023-09-12 08:16:02,946 "GET /users/accounts/logout/ HTTP/1.1" 302 0
INFO 2023-09-12 08:16:03,145 "GET / HTTP/1.1" 302 0
INFO 2023-09-12 08:16:03,673 "GET /accounts/login/?next=/ HTTP/1.1" 200 3985
INFO 2023-09-12 08:16:44,416 "POST /users/accounts/login/ HTTP/1.1" 302 0
INFO 2023-09-12 08:16:45,469 "GET / HTTP/1.1" 200 7052
INFO 2023-09-12 08:16:50,860 "GET /student/registered-courses/ HTTP/1.1" 200 9819
ERROR 2023-09-12 08:16:53,529 Internal Server Error: /student/course-material/7/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:16:53,530 "GET /student/course-material/7/ HTTP/1.1" 500 91749
INFO 2023-09-12 08:16:59,961 "GET /student/course-material/6/ HTTP/1.1" 200 10683
ERROR 2023-09-12 08:17:13,928 Internal Server Error: /student/course-material/7/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:17:13,928 "GET /student/course-material/7/ HTTP/1.1" 500 91749
ERROR 2023-09-12 08:18:37,909 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:18:37,909 "GET /student/course-material/8/ HTTP/1.1" 500 91749
INFO 2023-09-12 08:19:38,131 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:19:38,372 Watching for file changes with StatReloader
ERROR 2023-09-12 08:19:42,695 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:19:42,695 "GET /student/course-material/8/ HTTP/1.1" 500 92048
INFO 2023-09-12 08:20:01,757 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:20:01,992 Watching for file changes with StatReloader
INFO 2023-09-12 08:20:53,510 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:20:53,747 Watching for file changes with StatReloader
INFO 2023-09-12 08:26:15,289 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:26:15,518 Watching for file changes with StatReloader
INFO 2023-09-12 08:28:03,505 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:28:03,734 Watching for file changes with StatReloader
INFO 2023-09-12 08:28:23,965 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:28:24,181 Watching for file changes with StatReloader
INFO 2023-09-12 08:28:39,423 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:28:39,635 Watching for file changes with StatReloader
INFO 2023-09-12 08:28:54,890 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:28:55,117 Watching for file changes with StatReloader
INFO 2023-09-12 08:29:04,315 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:29:04,555 Watching for file changes with StatReloader
ERROR 2023-09-12 08:29:09,321 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 87, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:29:09,321 "GET /student/course-material/8/ HTTP/1.1" 500 92025
INFO 2023-09-12 08:29:24,906 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:29:25,127 Watching for file changes with StatReloader
ERROR 2023-09-12 08:29:29,524 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 87, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:29:29,525 "GET /student/course-material/8/ HTTP/1.1" 500 92027
INFO 2023-09-12 08:29:57,542 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:29:57,763 Watching for file changes with StatReloader
ERROR 2023-09-12 08:30:02,848 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:30:02,849 "GET /student/course-material/8/ HTTP/1.1" 500 92048
INFO 2023-09-12 08:31:14,353 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:31:14,576 Watching for file changes with StatReloader
ERROR 2023-09-12 08:31:19,354 Internal Server Error: /student/course-material/8/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 76, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:31:19,355 "GET /student/course-material/8/ HTTP/1.1" 500 92048
INFO 2023-09-12 08:31:27,852 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:31:28,078 Watching for file changes with StatReloader
INFO 2023-09-12 08:31:43,362 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:31:43,580 Watching for file changes with StatReloader
INFO 2023-09-12 08:33:42,964 "GET /student/course-material/6/ HTTP/1.1" 200 10603
INFO 2023-09-12 08:33:54,832 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:33:55,053 Watching for file changes with StatReloader
INFO 2023-09-12 08:34:05,262 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:34:05,482 Watching for file changes with StatReloader
INFO 2023-09-12 08:34:14,075 "GET /student/course-material/6/ HTTP/1.1" 200 10603
INFO 2023-09-12 08:34:15,073 "GET /student/course-material/6/ HTTP/1.1" 200 10603
INFO 2023-09-12 08:36:01,653 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:36:01,921 Watching for file changes with StatReloader
INFO 2023-09-12 08:36:14,098 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:36:14,349 Watching for file changes with StatReloader
ERROR 2023-09-12 08:37:24,934 Internal Server Error: /student/course-material/7/
Traceback (most recent call last):
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 135, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/contrib/auth/mixins.py", line 73, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/lanlokun/Desktop/Django-School-Management-System/venv/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
    context = self.get_context_data(object=self.object)
  File "/home/lanlokun/Desktop/portal/apps/students/views.py", line 78, in get_context_data
    course_materials = CourseMaterial.objects.filter(course=self.object.course)
AttributeError: 'NoneType' object has no attribute 'course'
ERROR 2023-09-12 08:37:24,935 "GET /student/course-material/7/ HTTP/1.1" 500 91799
INFO 2023-09-12 08:39:43,001 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:39:43,244 Watching for file changes with StatReloader
INFO 2023-09-12 08:39:52,260 "GET /student/course-material/7/ HTTP/1.1" 200 9304
INFO 2023-09-12 08:40:02,219 "GET /student/course-material/8/ HTTP/1.1" 200 9304
INFO 2023-09-12 08:40:07,616 "GET /student/course-material/9/ HTTP/1.1" 200 9304
INFO 2023-09-12 08:40:13,074 "GET /student/course-material/10/ HTTP/1.1" 200 9304
INFO 2023-09-12 08:40:26,766 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:40:27,020 Watching for file changes with StatReloader
INFO 2023-09-12 08:40:50,356 /home/lanlokun/Desktop/portal/apps/students/views.py changed, reloading.
INFO 2023-09-12 08:40:50,613 Watching for file changes with StatReloader
